In [2]:
import os
import pandas as pd

### FAO CROP

In [3]:
fao_Crops_Livestock_E_Europe = pd.read_csv("dataset/onedrive DS/fao_europe/Production_Crops_Livestock_E_Europe_cleaned.csv")

/tmp/ipykernel_1020/945253334.py:1: DtypeWarning: Columns (200) have mixed types. Specify dtype option on import or set low_memory=False.
  fao_Crops_Livestock_E_Europe = pd.read_csv("dataset/onedrive DS/fao_europe/Production_Crops_Livestock_E_Europe_cleaned.csv")


In [4]:
# on force les colonnes années en ligne
fao_Crops_Livestock_E_Europe = pd.melt(
    fao_Crops_Livestock_E_Europe, 
    id_vars=['Area', 'Item', 'Element', 'Unit'], # Les colonnes qu'on garde fixes
    value_vars=[col for col in fao_Crops_Livestock_E_Europe.columns if col.startswith('Y') and col[-1].isdigit()], # On cible les colonnes Y1961, Y1962...
    var_name='Year', # Le nom de la nouvelle colonne pour les années
    value_name='Value' # Le nom de la nouvelle colonne pour les scores
)
# Nettoyage rapide pour enlever le 'Y' devant l'année et la transformer en entier
fao_Crops_Livestock_E_Europe['Year'] = fao_Crops_Livestock_E_Europe['Year'].str.replace('Y', '').astype(int)

In [5]:
# on fitre sur les 39 area de l'europes
fao_Crops_Livestock_E_Europe = fao_Crops_Livestock_E_Europe.loc[fao_Crops_Livestock_E_Europe['Area'].isin(['Albania','Austria','Belarus','Belgium','Bosnia-Herzegovinia','Bulgaria','Croatia','Czech_Republic','Denmark','Estonia','Faeroes','Finland','France','Germany','Greece','Hungary','Iceland','Ireland','Italy','Latvia','Lithuania','Luxembourg','Malta','Montenegro','Netherlands','Macedonia','Norway','Poland','Portugal','Moldova','Romania','Serbia','Slovakia','Slovenia','Spain','Sweden','Switzerland','Ukraine','United_Kingdom']),:]


In [6]:
# on fitre sur l'éléments Yield
fao_Crops_Livestock_E_Europe = fao_Crops_Livestock_E_Europe.loc[fao_Crops_Livestock_E_Europe['Element']=='Yield',:]


In [7]:
# correspondance avec les autres domaines FAO pour être iso 'area'
mapping_fao_crop_to_europe = {
    'Czechia': 'Czech_Republic',
    'Faroe Islands': 'Faeroes',
    'Republic of Moldova': 'Moldova',
    'North Macedonia': 'Macedonia',
    'United Kingdom of Great Britain and Northern Ireland': 'United_Kingdom',
    'Netherlands (Kingdom of the)': 'Netherlands',
    'Bosnia and Herzegovina': 'Bosnia-Herzegovinia'     
}

In [8]:
fao_Crops_Livestock_E_Europe.loc[(fao_Crops_Livestock_E_Europe['Area'] == 'France') & (fao_Crops_Livestock_E_Europe['Item']=='Apples'),:]

,Area,Item,Element,Unit,Year,Value
4068,France,Apples,Yield,kg/ha,1961,22000.4
18140,France,Apples,Yield,kg/ha,1962,49054.8
32212,France,Apples,Yield,kg/ha,1963,24600.9
46284,France,Apples,Yield,kg/ha,1964,32013.2
60356,France,Apples,Yield,kg/ha,1965,34730.1
...,...,...,...,...,...,...
834316,France,Apples,Yield,kg/ha,2020,31865.5
848388,France,Apples,Yield,kg/ha,2021,30125.1
862460,France,Apples,Yield,kg/ha,2022,33055.5
876532,France,Apples,Yield,kg/ha,2023,35212.6


In [9]:
fao_Crops_Livestock_E_Europe.describe(include='all')

,Area,Item,Element,Unit,Year,Value
count,171648,171648,171648,171648,171648.000000,1.078230e+05
unique,39,132,1,2,NaN,NaN
top,Spain,"Roots and Tubers, Total",Yield,kg/ha,NaN,NaN
freq,7552,2496,171648,169024,NaN,NaN
mean,NaN,NaN,NaN,NaN,1992.500000,1.722886e+04
std,NaN,NaN,NaN,NaN,18.473007,1.973618e+05
min,NaN,NaN,NaN,NaN,1961.000000,0.000000e+00
25%,NaN,NaN,NaN,NaN,1976.750000,1.858200e+03
50%,NaN,NaN,NaN,NaN,1992.500000,5.000000e+03
75%,NaN,NaN,NaN,NaN,2008.250000,1.568275e+04


In [10]:
display(fao_Crops_Livestock_E_Europe.groupby('Area')['Area'].count())

Area
Albania                4608
Austria                4928
Belarus                3200
Belgium                3904
Bosnia-Herzegovinia    4224
Bulgaria               6528
Croatia                5568
Czech_Republic         5120
Denmark                3584
Estonia                3072
Faeroes                 128
Finland                3072
France                 6464
Germany                4608
Greece                 6656
Hungary                6336
Iceland                 704
Ireland                2624
Italy                  6656
Latvia                 3648
Lithuania              4608
Luxembourg             3264
Macedonia              4992
Malta                  3200
Moldova                4160
Montenegro             2816
Netherlands            3968
Norway                 2496
Poland                 5120
Portugal               6144
Romania                5632
Serbia                 4224
Slovakia               5440
Slovenia               5184
Spain                  7552
Sweden         

### FAO Fertiliz

In [11]:
fao_lis_fertiliz_data = pd.read_csv("dataset/fao_lis_fertiliz_data/fao_lis_fertiliz_data_5-18-2026.csv")

In [12]:
#mapping fao Crop pour les area
fao_lis_fertiliz_data['Area'] = fao_lis_fertiliz_data['Area'].replace(mapping_fao_crop_to_europe)

# on fitre sur les 39 area de l'europes
fao_lis_fertiliz_data = fao_lis_fertiliz_data.loc[fao_lis_fertiliz_data['Area'].isin(['Albania','Austria','Belarus','Belgium','Bosnia-Herzegovinia','Bulgaria','Croatia','Czech_Republic','Denmark','Estonia','Faeroes','Finland','France','Germany','Greece','Hungary','Iceland','Ireland','Italy','Latvia','Lithuania','Luxembourg','Malta','Montenegro','Netherlands','Macedonia','Norway','Poland','Portugal','Moldova','Romania','Serbia','Slovakia','Slovenia','Spain','Sweden','Switzerland','Ukraine','United_Kingdom']),:]


In [13]:
fao_lis_fertiliz_data.describe(include='all')

,Domain Code,Domain,Area Code (M49),Area,Element Code,Element,Item Code,Item,Year Code,Year,Unit,Value,Flag,Flag Description,Note
count,5567,5567,5567.000000,5567,5567.0,5567,5567.000000,5567,5567.000000,5567.000000,5567,5.567000e+03,5567,5567,4835
unique,1,1,NaN,39,NaN,1,NaN,3,NaN,NaN,1,NaN,4,4,5
top,RFN,Fertilizers by Nutrient,NaN,Iceland,NaN,Agricultural Use,NaN,Nutrient nitrogen N (total),NaN,NaN,t,NaN,A,Official figure,Official data from questionnaire
freq,5567,5567,NaN,189,NaN,5567,NaN,1865,NaN,NaN,5567,NaN,3731,3731,3437
mean,NaN,NaN,425.408838,NaN,5157.0,NaN,3102.996407,NaN,1996.371295,1996.371295,NaN,2.507580e+05,NaN,NaN,NaN
std,NaN,NaN,239.718712,NaN,0.0,NaN,0.816415,NaN,17.936114,17.936114,NaN,4.094504e+05,NaN,NaN,NaN
min,NaN,NaN,8.000000,NaN,5157.0,NaN,3102.000000,NaN,1961.000000,1961.000000,NaN,0.000000e+00,NaN,NaN,NaN
25%,NaN,NaN,246.000000,NaN,5157.0,NaN,3102.000000,NaN,1982.000000,1982.000000,NaN,2.506200e+04,NaN,NaN,NaN
50%,NaN,NaN,380.000000,NaN,5157.0,NaN,3103.000000,NaN,1999.000000,1999.000000,NaN,8.657400e+04,NaN,NaN,NaN
75%,NaN,NaN,620.000000,NaN,5157.0,NaN,3104.000000,NaN,2012.000000,2012.000000,NaN,2.922880e+05,NaN,NaN,NaN


In [14]:
display(fao_lis_fertiliz_data.groupby('Area')['Area'].count())

Area
Albania                184
Austria                188
Belarus                 96
Belgium                 66
Bosnia-Herzegovinia     87
Bulgaria               188
Croatia                 94
Czech_Republic          93
Denmark                189
Estonia                 96
Faeroes                 66
Finland                189
France                 189
Germany                189
Greece                 189
Hungary                189
Iceland                189
Ireland                189
Italy                  189
Latvia                  96
Lithuania               96
Luxembourg              66
Macedonia               91
Malta                  173
Moldova                 94
Montenegro              54
Netherlands            189
Norway                 189
Poland                 189
Portugal               189
Romania                189
Serbia                  54
Slovakia                93
Slovenia                94
Spain                  189
Sweden                 189
Switzerland            

### FAO Pihfr

In [15]:
fao_lis_pihfr_data = pd.read_csv("dataset/fao_lis_pihfr_data/fao_lis_pihfr_data_5-18-2026.csv")

In [16]:
#mapping fao Crop pour les area
fao_lis_pihfr_data['Area'] = fao_lis_pihfr_data['Area'].replace(mapping_fao_crop_to_europe)

# on fitre sur les 39 area de l'europes
fao_lis_pihfr_data = fao_lis_pihfr_data.loc[fao_lis_pihfr_data['Area'].isin(['Albania','Austria','Belarus','Belgium','Bosnia-Herzegovinia','Bulgaria','Croatia','Czech_Republic','Denmark','Estonia','Faeroes','Finland','France','Germany','Greece','Hungary','Iceland','Ireland','Italy','Latvia','Lithuania','Luxembourg','Malta','Montenegro','Netherlands','Macedonia','Norway','Poland','Portugal','Moldova','Romania','Serbia','Slovakia','Slovenia','Spain','Sweden','Switzerland','Ukraine','United_Kingdom']),:]


In [17]:
fao_lis_pihfr_data.head()

,Domain Code,Domain,Area Code (M49),Area,Element Code,Element,Item Code,Item,Year Code,Year,Unit,Value,Flag,Flag Description,Note
0,RP,Pesticides Use,8,Albania,5157,Agricultural Use,1357,Pesticides (total),1990,1990,t,121,I,Value imputed by a receiving agency,Imputed value
1,RP,Pesticides Use,8,Albania,5157,Agricultural Use,1357,Pesticides (total),1991,1991,t,121,I,Value imputed by a receiving agency,Imputed value
2,RP,Pesticides Use,8,Albania,5157,Agricultural Use,1357,Pesticides (total),1992,1992,t,121,I,Value imputed by a receiving agency,Imputed value
3,RP,Pesticides Use,8,Albania,5157,Agricultural Use,1357,Pesticides (total),1993,1993,t,121,A,Official figure,Official figure
4,RP,Pesticides Use,8,Albania,5157,Agricultural Use,1357,Pesticides (total),1994,1994,t,201,A,Official figure,Official figure


In [18]:
fao_lis_pihfr_data.describe(include='all')

,Domain Code,Domain,Area Code (M49),Area,Element Code,Element,Item Code,Item,Year Code,Year,Unit,Value,Flag,Flag Description,Note
count,6076,6076,6076.000000,6076,6076.0,6076,6076.000000,6076,6076.000000,6076.000000,6076,6076.000000,6076,6076,6076
unique,1,1,NaN,39,NaN,1,NaN,5,NaN,NaN,1,NaN,4,4,7
top,RP,Pesticides Use,NaN,Albania,NaN,Agricultural Use,NaN,Pesticides (total),NaN,NaN,t,NaN,A,Official figure,Official figure
freq,6076,6076,NaN,170,NaN,6076,NaN,1248,NaN,NaN,6076,NaN,3015,3015,2956
mean,NaN,NaN,419.418697,NaN,5157.0,NaN,1332.059908,NaN,2007.258558,2007.258558,NaN,4514.125741,NaN,NaN,NaN
std,NaN,NaN,240.999866,NaN,0.0,NaN,17.233932,NaN,9.561400,9.561400,NaN,11764.403605,NaN,NaN,NaN
min,NaN,NaN,8.000000,NaN,5157.0,NaN,1309.000000,NaN,1990.000000,1990.000000,NaN,0.000000,NaN,NaN,NaN
25%,NaN,NaN,233.000000,NaN,5157.0,NaN,1320.000000,NaN,1999.000000,1999.000000,NaN,24.000000,NaN,NaN,NaN
50%,NaN,NaN,380.000000,NaN,5157.0,NaN,1331.000000,NaN,2008.000000,2008.000000,NaN,619.500000,NaN,NaN,NaN
75%,NaN,NaN,620.000000,NaN,5157.0,NaN,1345.000000,NaN,2016.000000,2016.000000,NaN,2767.250000,NaN,NaN,NaN


### CRU


In [ ]:

# Configuration des chemins des dossiers contenant les données
path_base_cru = "dataset/onedrive DS/cru_data_selected"
liste_domaines = ['cld', 'pre', 'tmp', 'wet', 'frs', 'pet', 'dtr', 'tmx', 'tmn', 'vap']

fao_countries = [
    'Albania', 'Austria', 'Belarus', 'Belgium', 'Bosnia-Herzegovinia', 'Bulgaria',
    'Croatia', 'Czech_Republic', 'Denmark', 'Estonia', 'Faeroes', 'Finland', 
    'France', 'Germany', 'Greece', 'Hungary', 'Iceland', 'Ireland', 'Italy', 
    'Latvia', 'Lithuania', 'Luxembourg', 'Malta', 'Macedonia', 'Montenegro', 
    'Netherlands', 'Norway', 'Poland', 'Portugal', 'Moldova', 'Romania', 
    'Serbia', 'Slovakia', 'Slovenia', 'Spain', 'Sweden', 'Switzerland', 'Ukraine',
    'United_Kingdom'
]

# Dictionnaire de renommage de 'ANN' 
regles_ann = {
    'wet': 'wet_days_total',      # Fréquence
    'frs': 'frost_days_total',     # Fréquence
    'pre': 'pre_total_mm',        # Cumul    
    'pet': 'pet_total_mm',        # Cumul
    'tmp': 'tmp_mean_c',          # Moyenne
    'cld': 'cld_mean_pct',        # Moyenne
    'vap': 'vap_pressure_mean_hpa',# Moyenne
    'dtr': 'diurnal_range_mean_c', # Moyenne
    'tmx': 'tmp_max_mean_c',      # Moyenne
    'tmn': 'tmp_min_mean_c'       # Moyenne
}

df_cru_global = None

# Boucle principale par dossier de CRU (liste_domaines)
for domaine in liste_domaines:
    print(f"🔄 Traitement du sous-domaine : {domaine}...")
    path_domaine = os.path.join(path_base_cru, domaine)
    fichiers_du_dossier = os.listdir(path_domaine)
    
    # Liste pour stocker temporairement les DataFrames des 39 pays pour ce domaine
    df_domaine_pays = []
    # Boucle dans les fichiers du dossier "domaine"
    for country in fao_countries:
        # Recherche du nom dd pays dans la liste des fichiers
        matches = [f for f in fichiers_du_dossier if country.lower() in f.lower()]
        
        if matches:
            fichier_cible = matches[0]
            chemin_fichier = os.path.join(path_domaine, fichier_cible)
            
            df_res = pd.read_csv(chemin_fichier, sep=';', skiprows=3, na_values='-999.0')
            df_res.columns = df_res.columns.str.strip()
            
            # on ne garde que YEAR et ANN
            df_res = df_res[['YEAR', 'ANN']].copy()
            
            # Renommage de ANN avec le nom unique du domaine 
            nom_feature = regles_ann[domaine]
            df_res = df_res.rename(columns={'YEAR': 'Year', 'ANN': nom_feature})
            
            # création de la colonne pays/area FAO 
            df_res['Area'] = country
            
            df_domaine_pays.append(df_res)
            
    # On rassemble verticalement tous les pays pour ce domaine précis
    df_un_domaine_tous_pays = pd.concat(df_domaine_pays, ignore_index=True)
    
    # On fusionne horizontalement ce domaine avec les précédents sur (Area, Year)
    if df_cru_global is None:
        df_cru_global = df_un_domaine_tous_pays
    else:
        df_cru_global = pd.merge(df_cru_global, df_un_domaine_tous_pays, on=['Area', 'Year'], how='outer')



🔄 Traitement du sous-domaine : cld...
🔄 Traitement du sous-domaine : pre...
🔄 Traitement du sous-domaine : tmp...
🔄 Traitement du sous-domaine : wet...
🔄 Traitement du sous-domaine : frs...
🔄 Traitement du sous-domaine : pet...
🔄 Traitement du sous-domaine : dtr...
🔄 Traitement du sous-domaine : tmx...
🔄 Traitement du sous-domaine : tmn...
🔄 Traitement du sous-domaine : vap...


In [20]:
df_cru_global.shape

(4836, 12)

In [21]:
df_cru_global.describe(include='all')

,Year,cld_mean_pct,Area,pre_total_mm,tmp_mean_c,wet_days_total,frost_days_total,pet_total_mm,diurnal_range_mean_c,tmp_max_mean_c,tmp_min_mean_c,vap_pressure_mean_hpa
count,4836.000000,4836.000000,4836,4836.000000,4836.000000,4836.000000,4836.000000,4712.000000,4836.000000,4836.000000,4836.000000,4836.000000
unique,NaN,NaN,39,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,NaN,NaN,Albania,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,NaN,NaN,124,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,1962.500000,64.298511,NaN,832.354983,8.929508,159.990881,112.660463,1.944164,8.356328,13.129363,4.764082,9.497395
std,35.798254,8.566000,NaN,288.296599,3.778866,41.204574,49.602969,0.538428,1.451381,4.179055,3.496586,1.900581
min,1901.000000,35.900000,NaN,171.100000,-0.900000,36.800000,0.000000,0.700000,3.700000,2.600000,-5.100000,5.300000
25%,1931.750000,59.200000,NaN,618.800000,6.700000,131.300000,81.375000,1.600000,7.500000,10.600000,3.000000,8.500000
50%,1962.500000,64.900000,NaN,746.300000,9.000000,162.100000,112.750000,1.900000,8.400000,13.150000,4.800000,9.500000
75%,1993.250000,70.125000,NaN,1036.925000,10.700000,186.100000,141.900000,2.300000,9.400000,15.500000,6.300000,10.300000


In [22]:
df_cru_global.loc[df_cru_global['Year'] == 2024,:]

,Year,cld_mean_pct,Area,pre_total_mm,tmp_mean_c,wet_days_total,frost_days_total,pet_total_mm,diurnal_range_mean_c,tmp_max_mean_c,tmp_min_mean_c,vap_pressure_mean_hpa
123,2024,48.7,Albania,1064.6,14.7,124.3,47.7,3.0,10.2,19.8,9.6,10.9
247,2024,61.9,Austria,1153.5,9.4,177.1,111.9,2.1,8.9,13.9,5.0,9.2
371,2024,64.9,Belarus,646.5,9.2,154.6,121.0,2.0,9.0,13.7,4.7,10.0
495,2024,72.0,Belgium,1117.7,11.9,219.7,47.2,2.0,7.9,15.8,8.0,11.3
619,2024,55.6,Bosnia-Herzegovinia,913.0,13.0,141.1,72.1,2.6,10.3,18.2,7.8,10.5
743,2024,58.1,Bulgaria,574.8,13.4,86.7,77.7,2.8,11.1,19.0,7.9,10.8
867,2024,56.3,Croatia,1055.4,14.2,151.8,53.3,2.6,9.7,19.0,9.3,11.3
991,2024,67.4,Czech_Republic,715.9,10.6,159.4,92.6,2.3,9.0,15.1,6.0,9.9
1115,2024,64.3,Denmark,828.1,9.8,188.3,73.6,1.8,6.4,13.1,6.6,10.5
1239,2024,68.2,Estonia,638.4,7.6,173.7,130.0,1.6,7.3,11.3,4.0,9.6


### Merge Globale des 4 dataframes
 le merge des 4 dataframe (FAO CROP / FAO fertiliz / FAO pihfr / CRU)

df_cru_global

fao_lis_pihfr_data

fao_lis_fertiliz_data

fao_Crops_Livestock_E_Europe 

### fusion des 4 fichiers
	résultat 1247015 lignes au lieu des 67 000 estimées -> beaucoup de doublons
		📊 Crops      : Total lignes = 171648   | Lignes avec (Area, Year) répétées = 169152
		📊 Fertiliz   : Total lignes = 5567     | Lignes avec (Area, Year) répétées = 3701
		📊 Pihfr      : Total lignes = 6076     | Lignes avec (Area, Year) répétées = 4828
		📊 CRU Météo  : Total lignes = 4836     | Lignes avec (Area, Year) répétées = 0

#### même procédé que pour les colonnes années passées en ligne du fichier FAO Crop
-> bascule des lignes item en colonnes
8 nouvelles colonnes :
'Nutrient nitrogen N (total)', 'Nutrient phosphate P2O5 (total)',
       'Nutrient potash K2O (total)', 'Fungicides and Bactericides',
       'Herbicides', 'Insecticides', 'Pesticides (total)', 'Rodenticides'


In [ ]:
# Pivotement des données de nutriment pour passer les différents types en colonnes distinctes
fertiliz_pivot = fao_lis_fertiliz_data.pivot_table(
    index=['Area', 'Year'],  # Clé unique par ligne : un pays pour une année donnée
    columns='Item',          # Chaque type d'engrais devient un en-tête de colonne
    values='Value',
    aggfunc='first'
).reset_index()              # On remet Area et Year en colonnes standard au lieu de les laisser en index

# Même opération sur les données PIHFR (pesticides, etc.) pour aligner la structure
pihfr_pivot = fao_lis_pihfr_data.pivot_table(
    index=['Area', 'Year'], 
    columns='Item', 
    values='Value',
    aggfunc='first'
).reset_index()

In [ ]:
# Merge Final des 4 dataframes/domaines
# la jointure se fait par les années et pays
df_merge_1 = pd.merge(fao_Crops_Livestock_E_Europe, fertiliz_pivot, on=['Area', 'Year'], how='left')

df_merge_2 = pd.merge(df_merge_1, pihfr_pivot, on=['Area', 'Year'], how='left')

df_final_agritech = pd.merge(df_merge_2, df_cru_global, on=['Area', 'Year'], how='left')

# Vérification rapide de la dimension de la matrice 
print(f"Nombre de colonnes (Features dispo pour l'EDA) : {df_final_agritech.shape[1]}")

Nombre de colonnes (Features dispo pour l'EDA) : 24


### Filtre sur les années 2000 -> 2023

In [25]:
filtre_années =[2000,2001,2002,2003,2004,2005,2006,2007,2008,2009,2010,2011,2012,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022,2023]

df_final_agritech = df_final_agritech.loc[df_final_agritech['Year'].isin(filtre_années),:]

In [26]:
df_final_agritech.shape

(64368, 24)

In [27]:
df_final_agritech.info()

<class 'pandas.core.frame.DataFrame'>
Index: 64368 entries, 104598 to 168965
Data columns (total 24 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Area                             64368 non-null  object 
 1   Item                             64368 non-null  object 
 2   Element                          64368 non-null  object 
 3   Unit                             64368 non-null  object 
 4   Year                             64368 non-null  int64  
 5   Value                            55141 non-null  float64
 6   Nutrient nitrogen N (total)      63480 non-null  float64
 7   Nutrient phosphate P2O5 (total)  63480 non-null  float64
 8   Nutrient potash K2O (total)      63415 non-null  float64
 9   Fungicides and Bactericides      63708 non-null  float64
 10  Herbicides                       63708 non-null  float64
 11  Insecticides                     63708 non-null  float64
 12  Pesticides (total

In [28]:
chemin_export = "dataset/df_final_agritech.csv"

# Export en CSV
df_final_agritech.to_csv(chemin_export, index=False, encoding='utf-8')

In [29]:
print(os.path.abspath("dataset/df_final_agritech.csv"))

/mnt/g/Fullstack/modules/11 - Final Project/AgriTech/dataset/df_final_agritech.csv


### brouillon